quantum/debounce/eager_pk.c

```diff
@@ -88,16 +89,22 @@ void update_debounce_counters(uint8_t num_rows, uint8_t current_time) {
 
 // upload from raw_matrix to final matrix;
 void transfer_matrix_values(matrix_row_t raw[], matrix_row_t cooked[], uint8_t num_rows, uint8_t current_time) {
+  matrix_need_update = false;
   debounce_counter_t *debounce_pointer = debounce_counters;
   for (uint8_t row = 0; row < num_rows; row++) {
     matrix_row_t delta        = raw[row] ^ cooked[row];
     matrix_row_t existing_row = cooked[row];
     for (uint8_t col = 0; col < MATRIX_COLS; col++) {
       matrix_row_t col_mask = (ROW_SHIFTER << col);
-      if ((delta & col_mask) && *debounce_pointer == DEBOUNCE_ELAPSED) {
-        *debounce_pointer    = current_time;
-        counters_need_update = true;
-        existing_row ^= col_mask;  // flip the bit.
+      if (delta & col_mask) {
+        if (*debounce_pointer == DEBOUNCE_ELAPSED) {
+        counters_need_update = true;	          *debounce_pointer    = current_time;
```

Comment:
Style points available here.

```suggestion
          counters_need_update = true;
          *debounce_pointer    = current_time;
```

Reply:
Actually, this entire line needs to be removed, as it's redundant.

Reply:
LOL, I was only looking at the file deltas.